src/test/xmeta/test_xmeta_emit.cpp

```diff
@@ -283,16 +283,403 @@ std::vector<uint8_t> run_and_save_to_memory(std::istringstream & test_idl, std::
+    ExpectedProperty(const std::string &expectedName, const std::string &expectedType)
+        : ExpectedType(expectedName), type(ElementType::End), typeRef(expectedType) {}
+
+    void VerifyType(Property const& prop) const override
+    {
+        REQUIRE(prop.Name() == name);
+        REQUIRE(prop.Flags().value == method_attributes_no_flags().value);
+
+        if (type == ElementType::End)
+        {
+            VerifyTypeRefName(typeRef, prop.Parent().TypeNamespace(), std::get<coded_index<TypeDefOrRef>>(prop.Type().Type().Type()).TypeRef());
+        }
+        else
+        {
+            REQUIRE(prop.Type().Type().element_type() == type);
+        }
+
+        bool foundGetMethod = false;
+        bool foundSetMethod = false;
+        std::string getMethodName = "get_";
+        std::string setMethodName = "put_";
+        getMethodName += prop.Name();
+        setMethodName += prop.Name();
+        for (auto &&method : prop.Parent().MethodList())
+        {
+            if (method.Flags().SpecialName() == true)
+            {
+                if (getMethodName == method.Name())
+                {
+                    foundGetMethod = true;
+                }
+                else if (setMethodName == method.Name())
+                {
+                    foundSetMethod = true;
+                    REQUIRE(!method.Signature().ReturnType());
+                }
+            }
+        }
+        REQUIRE(foundGetMethod == true);
+        REQUIRE(foundSetMethod == true);
+    }
+};
+
+struct ExpectedEvent : ExpectedType {
+    std::string typeRef;
+
+    ExpectedEvent(const std::string &expectedName, const std::string &expectedType)
+        : ExpectedType(expectedName), typeRef(expectedType) {}
+
+    void VerifyType(Event const& event) const override
+    {
+        REQUIRE(event.Name() == name);
+        REQUIRE(event.EventFlags().value == event_attributes_no_flags().value);
+
+        VerifyTypeRefName(typeRef, event.Parent().TypeNamespace(), event.EventType().TypeRef());
+
+        bool foundAddMethod = false;
+        bool foundRemoveMethod = false;
+        std::string addMethodName = "add_";
+        std::string removeMethodName = "remove_";
+        addMethodName += event.Name();
+        removeMethodName += event.Name();
+        for (auto &&method : event.Parent().MethodList())
+        {
+            if (method.Flags().SpecialName() == true)
+            {
+                if (addMethodName == method.Name())
+                {
+                    foundAddMethod = true;
+                    REQUIRE(size(method.Signature().Params()) == 1);
+
+                    std::string expectedTypeName(std::get<coded_index<TypeDefOrRef>>(method.Signature().Params().first[0].Type().Type()).TypeRef().TypeName());
+                    VerifyTypeRefName(expectedTypeName, event.Parent().TypeNamespace(), event.EventType().TypeRef());
+                    REQUIRE(method.Flags().value == interface_method_event_attributes().value);
```

Comment:
Need to check the signature of the formal parameter. Particularly the add method has a return sig with type ref EventRegistrationToken and the remove method has formal parameter signature with type ref EventRegistrationToken. I don't see anything referring and checking for EventRegistrationToken. 

Look at this for example

    REQUIRE(add_method.Name() == "add_Changed");	
    {	
        auto const& sig = add_method.Signature();	
        REQUIRE(std::get<coded_index<TypeDefOrRef>>(sig.ReturnType().Type().Type()).TypeRef() == event_registration_token_ref);	
        REQUIRE(size(sig.Params()) == 1);	
        auto const& param_sig = sig.Params().first[0];	
        REQUIRE(std::get<coded_index<TypeDefOrRef>>(param_sig.Type().Type()).TypeRef() == string_list_event_ref);	
        REQUIRE(add_method.Flags().value == interface_method_event_attributes().value);	
    }	
    auto const& remove_method = event1.MethodSemantic().first[1].Method();	
    REQUIRE(remove_method.Name() == "remove_Changed");	
    {	    {
        auto const& sig = remove_method.Signature();	        std::make_unique<ExpectedInterface>("IComboBox",
        REQUIRE(!sig.ReturnType());	            std::vector<ExpectedEvent> {
        REQUIRE(size(sig.Params()) == 1);	                ExpectedEvent("Changed", "StringListEvent")
        auto const& param_sig = sig.Params().first[0];	            }
        REQUIRE(std::get<coded_index<TypeDefOrRef>>(param_sig.Type().Type()).TypeRef() == event_registration_token_ref);	        ),
        REQUIRE(remove_method.Flags().value == interface_method_event_attributes().value);	        std::make_unique<ExpectedDelegate>("StringListEvent", ElementType::Void,
    }